web/controllers/task_controller.ex

```diff
@@ -100,23 +103,36 @@ defmodule CoursePlanner.TaskController do
     end
   end
 
-  @error_messages %{
-    not_found: "Task was not found.",
-    already_finished: "Cannott grab as task is already finished.",
-    already_assigned: "Cannot grab as task is already assigned."
-  }
+  def grab(%{assigns: %{current_user: %{id: volunteer_id}}} = conn, %{"task_id" => task_id}) do
+    grab_and_drop_common_handling(conn, :grab, task_id, volunteer_id)
```

Comment:
what do you think of making `grab_and_drop_common_handling` just a response formatter? you can call the function that you want and pass the result to it instead of giving lots of params to be matched, for example
```elixir
Tasks.grab(task_id, volunteer_id) |> format_response(conn, "Task grabbed")
```